liota/dccs/iotcc.py

```diff
@@ -125,79 +124,34 @@ def on_receive_safe(msg):
-                    raise IOError('Could not open config file ' + fullPath)
-            except IOError, err:
-                log.error('Could not open config file')
-        else:
-            # missing config file
-            log.warn('liota.conf file missing')
-
-    def get_parent_reg_id(self, entity_obj):
-        config = ConfigParser.RawConfigParser()
-        fullPath = LiotaConfigPath().get_liota_fullpath()
-        if fullPath != '':
-            try:
-                if config.read(fullPath) != []:
-                    try:
-                        uuid_path = config.get('UUID_PATH', 'uuid_path')
-                        uuid_config = ConfigParser.RawConfigParser()
-                        uuid_config.optionxform = str
-                        uuid_config.read(uuid_path)
-                        if isinstance(entity_obj, System):
-                            entity_obj_name = "System"
-                        else:
-                            entity_obj_name = entity_obj.name
-                        stored_entity_id = uuid_config.get(entity_obj_name, "local_uuid")
-                        if stored_entity_id == entity_obj.entity_id:
-                            reg_parent_entity_id = uuid_config.get(entity_obj_name, "uuid")
-                        return reg_parent_entity_id
-                    except ConfigParser.ParsingError, err:
-                        log.error('Could not open config file')
-                else:
-                    raise IOError('Could not open config file ' + fullPath)
-            except IOError, err:
-                log.error('Could not open config file')
-        else:
-            # missing config file
-            log.warn('liota.conf file missing')
-
-    def _create_relationship(self, entity_parent_reg_id, entity_child_reg_id):
-        """ This function initializes all relations between System and it's children.
-            It is called after each object's UUID is received.
+    def create_relationship(self, reg_entity_parent, reg_entity_child):
+        """ This function initializes all relations between Registered Entities.
 
             Parameters:
             - obj: The object that has just obtained an UUID
         """
-        self.con.send(self._relationship(self.con.next_id(), entity_parent_reg_id, entity_child_reg_id))
+        # sanity check: must be RegisteredEntity or RegisteredMetricRegisteredMetric
+        if (not isinstance(reg_entity_parent, RegisteredEntity) \
+            and not isinstance(reg_entity_parent, RegisteredMetric)) \
+            or (not isinstance(reg_entity_child, RegisteredEntity) \
+                and not isinstance(reg_entity_child, RegisteredMetric)):
+            raise TypeError()
+
+        reg_entity_child.set_parent(reg_entity_parent)
+        if isinstance(reg_entity_child, RegisteredMetric):
+            # should save parent's reg_entity_id
+            reg_entity_child.set_reg_entity_id(reg_entity_parent.get_reg_entity_id())
```

Comment:
You don`t require getter & setter in python. Just directly call the object attribute.